src/security/forward.h

```diff
@@ -155,6 +163,14 @@ class ParsedOptions {}; // we never parse/use TLS options in this case
 
 class PeerConnector;
 class PeerOptions;
+
+#if USE_OPENSSL
+CtoCpp1(EVP_PKEY_free, EVP_PKEY *)
+typedef Security::LockingPointer<EVP_PKEY, EVP_PKEY_free_cpp, HardFun<int, EVP_PKEY *, EVP_PKEY_up_ref> > PrivateKeyPointer;
+#else
+typedef void *PrivateKeyPointer;
```

Comment:
As discussed in pr #38, this `PrivateKeyPointer` is not compatible with the primary `PrivateKeyPointer` declaration because the former lacks self-initialization. If possible, use a smart pointer here. The old [comment](https://github.com/squid-cache/squid/pull/38#issuecomment-320756541) from pr #38 has specific suggestions.

Reply:
You will have to convince the C++ committee to make deletion of void type valid syntax."Pointer<void> X = nullptr;" accepted syntax. Until then we cannot both initialize and have a smart void-pointer.

Reply:
> I'm against a piecewise approach for that fix.

I can accept that argument here if the fix is difficult. In that case, please consider adding an XXX. For example:

```
typedef void *PrivateKeyPointer; // XXX: incompatible with the other PrivateKeyPointer declaration (lacks self-initialization)
```

Reply:
Done.